### PR TITLE
Fixing crasher while printing error in some cases

### PR DESCRIPTION
**Pull request by bhariharan**

### CODE REVIEW

**Comment by huminzhi** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Code coverage under minimum of 80%</td>
    </tr>
  </tbody>
</table>

## Current coverage for SalesforceSDKCore.framework is `48%`
Files changed | - | - 
--- | --- | ---
SFRestAPI.m | `85%` | :white_check_mark:

---
## Current coverage for SalesforceAnalytics.framework is `41%`
✅ *No files affecting coverage found*

---

> Powered by [xcov](https://github.com/nakiostudio/xcov)

<p align="right" data-meta="generated_by_CI-result">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI.m [hunk 1]
```diff
@@ -273,10 +273,20 @@ - (void)replayRequestIfRequired:(NSData *)data response:(NSURLResponse *)respons
             }
         } else {
             if (!error) {
-                NSDictionary *errorDict = nil;
+                NSMutableDictionary *errorDict = nil;
```
Comment (khawkins) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI.m]: Minor: there doesn't appear to be a reason for this to be a mutable dictionary.  Could be a plain old `NSDictionary`.

**Comment by bhariharan** [comment]
Done [here](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/1988/commits/e2ff9008d2ceaf2a6512eb9aef9622959c682a5e).

**Comment by bhariharan** [comment]
Done [here](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/1988/commits/e2ff9008d2ceaf2a6512eb9aef9622959c682a5e).

**Pull request merged**